### DISCUSSION TITLE
Use Rook for local storage

### DISCUSSION THREAD

**Discussion by guettli** [rook/rook#14158] [General]
Is it possible to get local disk performance with rook?

It is perfectly fine to not have a replicated storage for this use case. 

We want to use cnPG. We tried topolvm and it works well.

Now I am curious: is it possible to get that performance with rook, too?

It's fine if this means the PVC consuming the volume is bound to a particular node. If a node and its storage fails, then this will be handled at a higher level. 

Since we plan to use Rook for s3 and RWX, it would be great, if we could use Rook for local storage, too.

What do you think?

**Comment by travisn** [comment]
My general recommendation for databases (that have their own replication) is to use local disks and not use Rook or Ceph. Ceph is designed for providing a network storage layer, and will not ever be as performant as the local disk, even if the OSD is on the same node as where your db pod is assigned. 

While Ceph with replica 1 is a possibility and is more performant than with higher replication, there are other implications. For example, if there are multiple OSDs in a replica 1 Ceph pool, the loss of any single OSD causes the loss of all data in that pool.

**Reply by guettli** [reply]
The implication that a replica count of one looses all data if that node goes down is fine. It's the same if we use local disks. The cnPG handles that via streaming replication and backups.

I had that idea because using rook also for that use-cases, would have the benefit the we have only one type of storage engine. At the moment we plan to have two: topolvm and rook
